Crash Report 3.3.1: java.lang.IllegalStateException FetchCoreUtils.getFileTempDir

```
Caused by: java.lang.IllegalStateException: 

 
  at com.tonyodev.fetch2core.FetchCoreUtils.getFileTempDir (FetchCoreUtils.java:100)

 
  at com.tonyodev.fetch2.FetchConfiguration$Builder.<init> (FetchConfiguration.java:64)

 
  at org.kiwix.kiwixmobile.core.di.modules.DownloaderModule.provideFetchConfiguration (DownloaderModule.java:71)

 
  at org.kiwix.kiwixmobile.core.di.modules.DownloaderModule_ProvideFetchConfigurationFactory.provideFetchConfiguration (DownloaderModule_ProvideFetchConfigurationFactory.java:48)

 
  at org.kiwix.kiwixmobile.core.di.modules.DownloaderModule_ProvideFetchConfigurationFactory.get (DownloaderModule_ProvideFetchConfigurationFactory.java:37)

 
  at org.kiwix.kiwixmobile.core.di.modules.DownloaderModule_ProvideFetchConfigurationFactory.get (DownloaderModule_ProvideFetchConfigurationFactory.java:12)

 
  at dagger.internal.DoubleCheck.get (DoubleCheck.java:47)

 
  at org.kiwix.kiwixmobile.core.di.modules.DownloaderModule_ProvideFetchFactory.get (DownloaderModule_ProvideFetchFactory.java:28)

 
  at org.kiwix.kiwixmobile.core.di.modules.DownloaderModule_ProvideFetchFactory.get (DownloaderModule_ProvideFetchFactory.java:10)

 
  at dagger.internal.DoubleCheck.get (DoubleCheck.java:47)

 
  at org.kiwix.kiwixmobile.core.downloader.fetch.FetchDownloadMonitor_Factory.get (FetchDownloadMonitor_Factory.java:30)

 
  at org.kiwix.kiwixmobile.core.downloader.fetch.FetchDownloadMonitor_Factory.get (FetchDownloadMonitor_Factory.java:9)

 
  at org.kiwix.kiwixmobile.core.di.modules.ApplicationModule_ProvideDownloadMonitor$core_releaseFactory.get (ApplicationModule_ProvideDownloadMonitor.java:31)

 
  at org.kiwix.kiwixmobile.core.di.modules.ApplicationModule_ProvideDownloadMonitor$core_releaseFactory.get (ApplicationModule_ProvideDownloadMonitor.java:10)

 
  at dagger.internal.DoubleCheck.get (DoubleCheck.java:47)

 
  at org.kiwix.kiwixmobile.core.di.components.DaggerCoreComponent.injectCoreApp (DaggerCoreComponent.java:414)

 
  at org.kiwix.kiwixmobile.core.di.components.DaggerCoreComponent.inject (DaggerCoreComponent.java:387)

 
  at org.kiwix.kiwixmobile.core.CoreApp.onCreate (CoreApp.java:90)
```

--- a/app/src/androidTest/java/org/kiwix/kiwixmobile/KiwixMockServer.kt
+++ b/app/src/androidTest/java/org/kiwix/kiwixmobile/KiwixMockServer.kt
@@ -38,11 +38,9 @@ class KiwixMockServer {
     mockWebServer.shutdown()
   }
 
-  fun map(
-    vararg pathsToResponses: Pair<String, Any>
-  ) {
+  fun map(vararg pathsToResponses: Pair<String, Any>) {
     val mapOfPathsToResponses = mapOf(*pathsToResponses)
-    mockWebServer.setDispatcher(object : Dispatcher() {
+    mockWebServer.dispatcher = object : Dispatcher() {
       override fun dispatch(request: RecordedRequest) =
         mapOfPathsToResponses[request.path]?.let(::successfulResponse)
           ?: forcedResponse?.let { return@let it }
@@ -50,7 +48,7 @@ class KiwixMockServer {
             "No response mapped for ${request.path}" +
               "\nmapped $mapOfPathsToResponses\nqueued $forcedResponse"
           )
-    })
+    }
   }
 
   private fun successfulResponse(bodyObject: Any) = MockResponse().apply {

--- a/buildSrc/src/main/kotlin/Versions.kt
+++ b/buildSrc/src/main/kotlin/Versions.kt
@@ -16,9 +16,9 @@ object Versions {
 
   const val androidx_test_espresso: String = "3.2.0" // available: "3.3.0"
 
-  const val com_squareup_retrofit2: String = "2.5.0" // available: "2.9.0"
+  const val com_squareup_retrofit2: String = "2.9.0"
 
-  const val com_squareup_okhttp3: String = "3.6.0" // available: "4.9.0"
+  const val com_squareup_okhttp3: String = "4.9.0"
 
   const val org_jetbrains_kotlin: String = "1.3.72" // available: "1.4.10"
 

--- a/buildSrc/src/main/kotlin/Versions.kt
+++ b/buildSrc/src/main/kotlin/Versions.kt
@@ -36,11 +36,11 @@ object Versions {
 
   const val org_jacoco: String = "0.7.9"
 
-  const val io_mockk: String = "1.10.0"
+  const val io_mockk: String = "1.10.0" // available: "1.10.2"
 
   const val android_arch_lifecycle_extensions: String = "1.1.1"
 
-  const val com_android_tools_build_gradle: String = "4.0.1"
+  const val com_android_tools_build_gradle: String = "4.0.1" // available: "4.0.2"
 
   const val de_fayard_buildsrcversions_gradle_plugin: String = "0.7.0"
 
@@ -50,9 +50,9 @@ object Versions {
 
   const val ink_page_indicator: String = "1.3.0"
 
-  const val leakcanary_android: String = "2.4"
+  const val leakcanary_android: String = "2.4" // available: "2.5"
 
-  const val constraintlayout: String = "1.1.3" // available: "2.0.1"
+  const val constraintlayout: String = "1.1.3" // available: "2.0.2"
 
   const val collection_ktx: String = "1.1.0"
 
@@ -60,7 +60,7 @@ object Versions {
 
   const val junit_jupiter: String = "5.6.2" // available: "5.7.0"
 
-  const val xfetch2okhttp: String = "3.1.4" // available: "3.1.5"
+  const val xfetch2okhttp: String = "3.1.5"
 
   const val assertj_core: String = "3.16.1" // available: "3.17.2"
 
@@ -68,7 +68,7 @@ object Versions {
 
   const val fragment_ktx: String = "1.2.5"
 
-  const val lint_gradle: String = "27.0.1"
+  const val lint_gradle: String = "27.0.1" // available: "27.0.2"
 
   const val testing_ktx: String = "1.1.2"
 
@@ -84,7 +84,7 @@ object Versions {
 
   const val rxandroid: String = "2.1.1"
 
-  const val core_ktx: String = "1.3.1"
+  const val core_ktx: String = "1.3.1" // available: "1.3.2"
 
   const val kiwixlib: String = "9.4.0"
 
@@ -92,19 +92,19 @@ object Versions {
 
   const val multidex: String = "2.0.1"
 
-  const val barista: String = "2.7.1" // available: "3.6.0"
+  const val barista: String = "2.7.1" // available: "3.7.0"
 
-  const val xfetch2: String = "3.1.4" // available: "3.1.5"
+  const val xfetch2: String = "3.1.5"
 
   const val jsr305: String = "3.0.2"
 
   const val ktlint: String = "0.36.0" // available: "0.39.0"
 
-  const val rxjava: String = "2.2.19"
+  const val rxjava: String = "2.2.19" // available: "2.2.20"
 
   const val webkit: String = "1.2.0" // available: "1.3.0"
 
-  const val aapt2: String = "4.0.1-6197926"
+  const val aapt2: String = "4.0.1-6197926" // available: "4.0.2-6197926"
 
   const val junit: String = "1.1.1" // available: "1.1.2"
 

